src/legacy/core_plugins/tagcloud/public/tag_cloud_fn.ts

```diff
@@ -17,72 +17,104 @@
  * under the License.
  */
 
-import { functionsRegistry } from 'plugins/interpreter/registries';
 import { i18n } from '@kbn/i18n';
 
-export const tagcloud = () => ({
-  name: 'tagcloud',
+import { ExpressionFunction, KibanaDatatable, Render } from '../../interpreter/types';
+
+const name = 'tagcloud';
+
+type Context = KibanaDatatable;
+
+interface Arguments {
+  scale: string;
+  orientation: string;
+  minFontSize: number;
+  maxFontSize: number;
+  showLabel: boolean;
+  metric: any; // these aren't typed yet
+  bucket: any; // these aren't typed yet
+}
+
+type VisParams = Omit<Arguments, 'bucket'>;
+
+interface RenderValue {
+  visType: typeof name;
+  visData: Context;
+  visConfig: VisParams;
+  params: any;
+}
+
+type Return = Render<RenderValue>;
+
+export const createTagCloudFn = (): ExpressionFunction<
+  typeof name,
+  Context,
+  Arguments,
+  Return
+> => ({
+  name,
   type: 'render',
   context: {
-    types: [
-      'kibana_datatable'
-    ],
+    types: ['kibana_datatable'],
   },
   help: i18n.translate('tagCloud.function.help', {
-    defaultMessage: 'Tagcloud visualization'
+    defaultMessage: 'Tagcloud visualization',
   }),
   args: {
     scale: {
       types: ['string'],
       default: 'linear',
       options: ['linear', 'log', 'square root'],
       help: i18n.translate('tagCloud.function.scale.help', {
-        defaultMessage: 'Scale to determine font size of a word'
+        defaultMessage: 'Scale to determine font size of a word',
       }),
     },
     orientation: {
       types: ['string'],
       default: 'single',
       options: ['single', 'right angled', 'multiple'],
       help: i18n.translate('tagCloud.function.orientation.help', {
-        defaultMessage: 'Orientation of words inside tagcloud'
+        defaultMessage: 'Orientation of words inside tagcloud',
       }),
     },
     minFontSize: {
       types: ['number'],
       default: 18,
+      help: '',
     },
     maxFontSize: {
       types: ['number'],
-      default: 72
+      default: 72,
+      help: '',
     },
     showLabel: {
       types: ['boolean'],
       default: true,
+      help: '',
     },
     metric: {
       types: ['vis_dimension'],
       help: i18n.translate('tagCloud.function.metric.help', {
-        defaultMessage: 'metric dimension configuration'
+        defaultMessage: 'metric dimension configuration',
       }),
       required: true,
     },
     bucket: {
       types: ['vis_dimension'],
       help: i18n.translate('tagCloud.function.bucket.help', {
-        defaultMessage: 'bucket dimension configuration'
+        defaultMessage: 'bucket dimension configuration',
       }),
     },
   },
-  fn(context, args) {
+  fn(context: Context, args: Arguments) {
```

Comment:
nit: I didn't catch this before, but `Context` and `Arguments` are not necessary to include here... they should already be enforced by the `ExpressionFunction` generic.